packages/compass/src/main/menu.ts

```diff
@@ -24,12 +24,34 @@ function separator(): MenuItemConstructorOptions {
   };
 }
 
-function quitItem(label: string): MenuItemConstructorOptions {
+function quitItem(
+  label: string,
+  compassApp: typeof CompassApplication
+): MenuItemConstructorOptions {
   return {
     label: label,
     accelerator: 'CmdOrCtrl+Q',
     click() {
-      app.quit();
+      !compassApp.preferences.getPreferences().enableShowDialogOnQuit
+        ? app.quit()
+        : void dialog
+            .showMessageBox({
+              type: 'warning',
+              title: 'Quit ' + app.getName(),
```

Comment:
We typically prefer to use the javascript string template literals with the backtick character ` for building strings in Compass
```suggestion
              title: `Quit ${app.getName()}`,
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals